### PR TITLE
[ASV-572] horizontal scroll trending bundles

### PR DESCRIPTION
**Pull request by JDSM01**
**What does this PR do?**

  Adds horizontal scroll to bundles that have other bundles.
  Removes bundles from the list that don't have any app in it.

**Questions**

AdultContentFilter is not being used in StoreTabFragmentChooser at the moment. I researched where it was being passed as true and there were 2 places.
One of them I'm almost sure it's not being used anymore, in StorePagerAdapter (search for API).
The other one is in DeeplinkManager in GenericDeeplink.
This boolean is used to create a displayable of a switch 
Do you know if these are really being used specially the second one?

**Database changed?**

   No

**Where should the reviewer start?**

RemoteBundleDataSource.java
BundleRepository.java
MoreBundleFragment.java
MoreBundlePresenter.java

**How should this be manually tested?**

Navigate trough the Trending and Top Games and other bundles that have more bundles inside them
Check if Home is also working correctly

**What are the relevant tickets?**

https://aptoide.atlassian.net/browse/ASV-572 (task)
https://aptoide.atlassian.net/browse/ASV-311 (user-story)

**Code Review Checklist**

- [ ] Architecture
- [ ] Documentation on public interfaces
- [ ] Database changed?
- [ ] If yes - Database migration?
- [ ] Remove comments & unused code & forgotten testing Logs
- [ ] Codestyle
- [ ] Partners build
- [ ] Unit tests pass
- [ ] Functional QA tests pass

### CODE REVIEW

**Comment by jdandrade** [comment]
There is a bug regarding load more:
Trending more > applications more > pull to refresh > scroll till end > (load more) > scroll till end > (end reached) > if you go up and down you can load more items again, meaning the end was not reached last time.

**Comment by jdandrade** [comment]
There is also a bug inside stores view, which is being parsed as the same fragment as the more one. We have to take in account that also.

File: app/src/main/java/cm/aptoide/pt/home/BundlesResponseMapper.java [hunk 1]
```diff
@@ -52,6 +52,10 @@ public BundlesResponseMapper(String marketName, InstallManager installManager,
         Object viewObject = widget.getViewObject();
         String title = widget.getTitle();
         if (type.equals(HomeBundle.BundleType.APPS) || type.equals(HomeBundle.BundleType.EDITORS)) {
+          if (event != null && event.getName()
+              .equals(Event.Name.getStoreWidgets)) {
+            event.setName(Event.Name.getMoreBundle);
+          }
```
Comment (JDSM01) [inline app/src/main/java/cm/aptoide/pt/home/BundlesResponseMapper.java]: Maybe this change of eventName can go before the ifs of the type of bundles

File: app/src/main/java/cm/aptoide/pt/home/FakeBundleDataSource.java [hunk 2]
```diff
@@ -13,19 +13,29 @@
...
   }
 
-  @Override public boolean hasMore(Integer offset) {
+  @Override public boolean hasMore(Integer offset, String title) {
     return true;
   }
 
-  public Single<HomeBundlesModel> getHomeBundles() {
+  @Override public Single<HomeBundlesModel> loadFreshBundleForEvent(String url, String key) {
+    return null;
```
Comment (jdandrade) [inline app/src/main/java/cm/aptoide/pt/home/FakeBundleDataSource.java]: This should return the getHomeBundles model, can be useful for testing for example.

File: app/src/main/java/cm/aptoide/pt/home/FakeBundleDataSource.java [hunk 3]
```diff
@@ -13,19 +13,29 @@
...
+  }
+
+  @Override
+  public Single<HomeBundlesModel> loadNextBundleForEvent(String url, int offset, String key,
+      int limit) {
+    return null;
```
Comment (jdandrade) [inline app/src/main/java/cm/aptoide/pt/home/FakeBundleDataSource.java]: This should return the getHomeBundles model, can be useful for testing for example.

File: app/src/vanilla/java/cm/aptoide/pt/app/view/MoreBundleView.java [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+import cm.aptoide.pt.home.HomeEvent;
+import cm.aptoide.pt.presenter.View;
+import java.util.List;
+import rx.Observable;
+
+/**
+ * Created by D01 on 05/06/2018.
+ */
+
+public interface MoreBundleView extends View {
```
Comment (jdandrade) [inline app/src/vanilla/java/cm/aptoide/pt/app/view/MoreBundleView.java]: This interface already exists (homeview) we should probably just add the toolbarinfo method and extend the homeview (but maybe change the "homeview" to bundles view or something)

**Pull request merged**